kernel/src/process/process_vm/init_stack/mod.rs

```diff
@@ -42,14 +42,17 @@ pub mod aux_vec;
 /// Set the initial stack size to 8 megabytes, following the default Linux stack size limit.
 pub const INIT_STACK_SIZE: usize = 8 * 1024 * 1024; // 8 MB
 
-/// The max number of arguments that can be used to creating a new process.
-pub const MAX_ARGV_NUMBER: usize = 128;
-/// The max number of environmental variables that can be used to creating a new process.
-pub const MAX_ENVP_NUMBER: usize = 128;
-/// The max length of each argument to create a new process.
-pub const MAX_ARG_LEN: usize = 2048;
-/// The max length of each environmental variable (the total length of key-value pair) to create a new process.
-pub const MAX_ENV_LEN: usize = 128;
+/// Maximum number of argument or environment strings that can be supplied to
+/// the `execve` system call.
+///
+/// Reference: <https://elixir.bootlin.com/linux/v6.15/source/include/uapi/linux/binfmts.h#L15>
+pub const MAX_ARG_STRING_NUM: usize = 0x7fff_ffff;
```

Comment:
`MAX_NR_STRING_ARGS`, which directly translates to "the maximum number of string arguments", is a more readable name.

`u32::MAX as usize` would be more readable than `0x7fff_ffff`.